activerecord/lib/active_record/relation/batches.rb

```diff
@@ -373,32 +373,44 @@ def batch_on_unloaded_relation(relation:, start:, finish:, load:, order:, use_ra
           if load
             records = batch_relation.records
             ids = records.map(&:id)
+            ids_size = ids.size
+            ids_last = ids.last
```

Comment:
@maximerety are we guaranteed that ids are sorted? or it doesn't matter here?

Reply:
These ids are already sorted indeed, because the `batch_relation` used is reordered with `reorder(batch_orders.to_h)` at the beginning of the method.

Note that I'm merely caching some values earlier here, but I did not change the logic of checking them later in the implementation. So it was already expected that the list was sorted.